addons/binding/org.openhab.binding.neato/src/main/java/org/openhab/binding/neato/handler/NeatoHandler.java

```diff
@@ -0,0 +1,257 @@
+
+    private Logger logger = LoggerFactory.getLogger(NeatoHandler.class);
+
+    private String vacuumSerialNumber;
+    private String vacuumSecret;
+    private String vacuumName;
+
+    private NeatoRobot mrRobot;
+
+    private int refreshTime;
+    private ScheduledFuture<?> refreshTask;
+    private static int DEFAULTREFRESHTIME = 60;
+
+    private ThingStatus thisStatus;
+
+    public NeatoHandler(Thing thing) {
+        super(thing);
+        thisStatus = ThingStatus.OFFLINE;
+    }
+
+    public ScheduledFuture<?> getRefreshTask() {
+        return this.refreshTask;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(org.openhab.binding.neato.NeatoBindingConstants.COMMAND)) {
+            logger.debug("Ok - will handle command for CHANNEL_COMMAND");
+
+            try {
+                mrRobot.sendCommand(command);
+            } catch (Exception exc) {
+                logger.error("Error while processing command from openHAB. Error: {}", exc.getMessage());
+            }
+
+        }
+        this.refreshStateAndUpdate();
+    }
+
+    @Override
+    public void initialize() {
+
+        logger.info("Will boot up Neato Vacuum Cleaner binding!");
+
+        updateStatus(ThingStatus.OFFLINE);
+        thisStatus = ThingStatus.OFFLINE;
+
+        Configuration config = getThing().getConfiguration();
+        vacuumSecret = (String) config.get(CONFIG_SECRET);
+        vacuumName = (String) config.get(CONFIG_NAME);
+        vacuumSerialNumber = (String) config.get(CONFIG_SERIAL);
+
+        logger.info("Got config settings for {} with serial number: {}", vacuumName, vacuumSecret);
+
+        super.initialize();
+
+        try {
+            refreshTime = ((BigDecimal) config.get(CONFIG_REFRESHTIME)).intValue();
+            if (refreshTime <= 0) {
+                throw new IllegalArgumentException("Refresh time must be positive number!");
+            }
+        } catch (Exception e) {
+            logger.warn("Refresh time [{}] is not valid. Falling back to default value: {}. {}", refreshTime,
+                    DEFAULTREFRESHTIME, e.getMessage());
+            refreshTime = DEFAULTREFRESHTIME;
+        }
+
+        mrRobot = new NeatoRobot(vacuumSerialNumber, vacuumSecret, vacuumName);
+        try {
+
+            this.startAutomaticRefresh();
+
+        } catch (Exception ex) {
+            // Note: When initialization can NOT be done set the status with more details for further
+            // analysis. See also class ThingStatusDetail for all available status details.
+            // Add a description to give user information to understand why thing does not work
+            // as expected. E.g.
+            // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+            // "Can not access device as username and/or password are invalid");
+
+            logger.error(ex.getMessage());
```

Comment:
Use a context (where are we what exception are we catching) and placeholder. Also don't log this as an error, you already updated the status, so it should at max be a warn.